src/scrabble.js

```diff
@@ -1,5 +1,191 @@
-function scrabble() {
+function normalScore(aWord) {
+  if (aWord === '' || aWord === ' \t\n' || aWord === null) {
+    return 0
+  } else {
+    const newWord = aWord
+    console.log('new word is', newWord)
+    let sum = 0
+    for (i = 0; i < newWord.length; i++) {
+      if (
+        newWord[i].includes('A') ||
+        newWord[i].includes('E') ||
+        newWord[i].includes('I') ||
+        newWord[i].includes('O') ||
+        newWord[i].includes('U') ||
+        newWord[i].includes('L') ||
+        newWord[i].includes('N') ||
+        newWord[i].includes('R') ||
+        newWord[i].includes('S') ||
+        newWord[i].includes('T') ||
+        newWord[i].includes('a') ||
+        newWord[i].includes('e') ||
+        newWord[i].includes('i') ||
+        newWord[i].includes('o') ||
+        newWord[i].includes('u') ||
+        newWord[i].includes('l') ||
+        newWord[i].includes('n') ||
+        newWord[i].includes('r') ||
+        newWord[i].includes('s') ||
+        newWord[i].includes('t')
+      ) {
+        sum += 1
+      }
+      if (
+        newWord[i].includes('D') ||
+        newWord[i].includes('G') ||
+        newWord[i].includes('d') ||
+        newWord[i].includes('g')
+      ) {
+        sum += 2
+      }
+      if (
+        newWord[i].includes('B') ||
+        newWord[i].includes('C') ||
+        newWord[i].includes('M') ||
+        newWord[i].includes('P') ||
+        newWord[i].includes('b') ||
+        newWord[i].includes('c') ||
+        newWord[i].includes('m') ||
+        newWord[i].includes('p')
+      ) {
+        sum += 3
+      }
+      if (
+        newWord[i].includes('F') ||
+        newWord[i].includes('H') ||
+        newWord[i].includes('V') ||
+        newWord[i].includes('W') ||
+        newWord[i].includes('Y') ||
+        newWord[i].includes('f') ||
+        newWord[i].includes('h') ||
+        newWord[i].includes('v') ||
+        newWord[i].includes('w') ||
+        newWord[i].includes('y')
+      ) {
+        sum += 4
+      }
+      if (newWord[i].includes('K') || newWord[i].includes('k')) {
+        sum += 5
+      }
+      if (
+        newWord[i].includes('J') ||
+        newWord[i].includes('X') ||
+        newWord[i].includes('j') ||
+        newWord[i].includes('x')
+      ) {
+        sum += 8
+      }
+      if (
+        newWord[i].includes('Q') ||
+        newWord[i].includes('Z') ||
+        newWord[i].includes('q') ||
+        newWord[i].includes('z')
+      ) {
+        sum += 10
+      }
+    }
+    return sum
+  }
+}
+
+function doubleLetter(aWord) {
+  const regex1 = /{[A-Z|a-z]}/g
+  const found1 = aWord.match(regex1)
+  const doubleLetter = normalScore(found1) * 2
+  return doubleLetter
+}
+
+console.log('Double letter test', doubleLetter('{A}{A}'))
+
+function trippleLetter(aWord) {
+  const regex2 = /\[[A-Z | a-z]]/g
+  const found2 = aWord.match(regex2)
+  const trippleLetter = normalScore(found2) * 3
+  return trippleLetter
+}
+
+function normalLetter(aWord) {
```

Comment:
I'm not 100% sure on what this function does, from line 107 to 113. Any chance you can reply to this comment and explain it for me?

Reply:
This was to clean up the argument and to avoid double scoring.

line 108 I look for anything inside { } and replace with ' ' and the same for the letters in the squares brackets[ ]

So for example in the word H{A}NDLE it would look for the {A} in line 108 and replace it with empty.

then return the score for normal letters using a clean argument of HNDLE.